backend/src/models/projects/budget.js

```diff
@@ -0,0 +1,100 @@
+const dbConnection = require("@database/databaseConnection");
+const { knex, dataBaseSchemas } = dbConnection();
+const projectBudgetTable = `${dataBaseSchemas().data}.project_budget`;
+const fiscalYearTable = `${dataBaseSchemas().data}.fiscal_year`;
+const projectDeliverableTable = `${dataBaseSchemas().data}.project_deliverable`;
+const contractsTable = `${dataBaseSchemas().data}.contract`;
+const recoveriesTable = `${dataBaseSchemas().data}.recovery_type`;
+const portfolioTable = `${dataBaseSchemas().data}.portfolio`;
+const clientCodingTable = `${dataBaseSchemas().data}.client_coding`;
+
+const findAllById = (projectId) => {
+  return knex(`${projectBudgetTable} as prb`)
+    .columns(
+      "prb.id",
+      knex.raw("prb.q1_amount::numeric::float8"),
+      "prb.q1_recovered",
+      knex.raw("prb.q2_amount::numeric::float8"),
+      "prb.q2_recovered",
+      knex.raw("prb.q3_amount::numeric::float8"),
+      "prb.q3_recovered",
+      knex.raw("prb.q4_amount::numeric::float8"),
+      "prb.q4_recovered",
+      "fy.fiscal_year as fiscal",
+      "prb.notes",
+      "prd.deliverable_name",
+      knex.raw("prb.detail_amount::numeric::float8"),
+      "rec.recovery_type_name",
+      "prb.stob",
+      "prb.client_coding_id", //TODO Add a lookup in the future.
+      "cntr.co_number"
+    )
+    .leftJoin(`${fiscalYearTable} as fy`, { "prb.fiscal": `fy.id` })
+    .leftJoin(`${projectDeliverableTable} as prd`, { "prb.project_deliverable_id": "prd.id" })
+    .leftJoin(`${recoveriesTable} as rec`, { "prb.recovery_area": "rec.id" })
+    .leftJoin(`${contractsTable} as cntr`, { "prb.contract_id": "cntr.id" })
+    .orderBy("prb.id")
+    .where({ "prd.project_id": projectId });
+};
+
+// Get specific one by id.
+const findById = (id) => {
+  return knex
+    .select(
+      "prb.id",
+      knex.raw("prb.q1_amount::numeric::float8"),
+      "prb.q1_recovered",
+      knex.raw("prb.q2_amount::numeric::float8"),
+      "prb.q2_recovered",
+      knex.raw("prb.q3_amount::numeric::float8"),
+      "prb.q3_recovered",
+      knex.raw("prb.q4_amount::numeric::float8"),
+      "prb.q4_recovered",
+      knex.raw(
+        "(SELECT json_build_object('value', prb.fiscal, 'label', COALESCE(fy.fiscal_year, ''))) AS fiscal"
+      ),
+      "prb.notes",
+      knex.raw(
+        "(SELECT json_build_object('value', prb.project_deliverable_id, 'label', COALESCE(prd.deliverable_name, ''))) AS project_deliverable_id"
+      ),
+      knex.raw("prb.detail_amount::numeric::float8"),
+      knex.raw(
+        "(SELECT json_build_object('value', prb.recovery_area, 'label', COALESCE(port.portfolio_name, ''))) AS recovery_area"
+      ),
+      knex.raw(
+        "(SELECT json_build_object('value', prb.resource_type, 'label', COALESCE(prb.resource_type, ''))) AS resource_type"
+      ),
+      "prb.stob",
+      knex.raw(
+        "(SELECT json_build_object('value', prb.client_coding_id, 'label', COALESCE(cc.program_area, ''))) AS client_coding_id"
```

Comment:
```suggestion
        "(SELECT json_build_object('value', prb.client_coding_id, 'label', COALESCE(cc.program_area, cc.client))) AS client_coding_id"
```
If cc.program_area is null then this will be {value: [id], label: ""} but the actual picker option (from the picker_options model line 198) would be {value: [id], label: [cc.client]} so they wouldn't match even though it is the same value.